### PR TITLE
Style: resolve #280 (clang-format comments, filename extension, mixed-case filenames)

### LINKED ISSUE #280: Style refactor all .cpp filenames + adjust license formatting for clang-format
**By submitting this issue, I confirm the following:**

- I have read and understood the [contributor guide](https://github.com/monero-project/kovri/blob/master/doc/CONTRIBUTING.md).
- I have checked that the issue I am reporting can be replicated or that the feature I am suggesting is not present.
- I have checked opened or recently closed [pull requests](https://github.com/monero-project/kovri/pulls) for existing solutions/implementations to my issue/suggestion.

*Place an X inside the bracket to confirm*
- [X] I confirm.

---

1. To keep in order with our upstream guide, we should rename all files from ```.cpp``` to ```.cc``` and change from ```MixedCase.cpp``` to ```single_case_with_underscore.cc```. Once that's done, remove the line about filename style in our style guide.

2. Because our column limit for clang-format is 80, clang-format will squish the license in formatted files. To avoid this, we need to amend a ```//``` at the end of each line like so:
 
```cpp
/**                                                                                       //
 * Copyright (c) 2015-2016, The Kovri I2P Router Project                                  //
 *                                                                                        //
 * All rights reserved.                                                                   //
 *                                                                                        //
 * Redistribution and use in source and binary forms, with or without modification, are   //
 * permitted provided that the following conditions are met:                              //
 *                                                                                        //
 * 1. Redistributions of source code must retain the above copyright notice, this list of //
 *    conditions and the following disclaimer.                                            //
 ```

or, amend ```clang-format on``` ```clang-format off``` before and after the license which I think is uglier and intrusive.  I propose we stick with the less-intrusive version of ```//```. If there is a better way to accomplish 2., I'm all ears.

Referencing #279 and #238.

### PR DESCRIPTION
**Pull request by guzzijones**
**By submitting this pull-request, I confirm the following:**

- I have read and understood the [Contributing Guide](https://github.com/monero-project/kovri/blob/master/doc/CONTRIBUTING.md).
- I have checked that another pull-request for this purpose does not exist.
- I have considered and confirmed that this submission will be valuable to others.
- I accept that this submission may not be used and that this pull-request may be closed by the will of the maintainer.
- I give this submission freely under the BSD 3-clause license.

*Place an X inside the bracket (or click the box in preview) to confirm*
- [x] I confirm.

---

*Enter your pull-request summary here*

adjusted license comment
renamed all .cpp files
remove mv.sh
fixed CMakeLists.txt files with new .cc
removed t.txt
fixed comments.  appended // to original comments
missed util files

### CODE REVIEW

**Comment by guzzijones** [comment]
I saw your note on 280.  I should be able to write a quick python script to take care of that one, too.

**Comment by anonimal** [comment]
Hi @guzzijones, I've opened https://github.com/guzzijones/kovri/pull/1

Per our [contributing guide](https://github.com/monero-project/kovri/blob/master/doc/CONTRIBUTING.md), for the remainder of the work, there's no need to squash any commits:

```
&anonimal | guzzi: if I may offer a workflow suggestion:
&anonimal | from our contributing guide:
&anonimal | "commits should be [atomic](https://en.wikipedia.org/wiki/Atomic_commit#Atomic_commit_convention) and diffs should be easy to read. For this reason, please try to not mix formatting fixes with non-formatting commits." 
&anonimal | e.g., one commit for appending // at the end of license, one commit for moving filenames, one commit for amending cmake with new names, etc.
```
JFTR: you don't have to close this PR to include new commits (if possible, please don't). Simply merge my PR and push your remaining committed work to your ```comments``` branch and github will take care of the rest.

Thanks again @guzzijones, I look forward to the rest :smile:

File: doc/STYLE.md [hunk 1]
```diff
@@ -67,5 +66,4 @@ return SSUPacket::GetSize()
 - Avoid prepended mixed-case ```k``` and MACRO_TYPE for all constants
 - Use Doxygen three-slash ```/// C++ comments``` when documenting for Doxygen
 - Document all your work for Doxygen as you progress
-- New files should maintain consistency with other filename style
```
Comment (EinMByte) [inline doc/STYLE.md]: This should be kept.

**Comment by anonimal** [comment]
I'm going to merge this and follow up with an immediate fix for the failed build (one filename was missed in tests).

**Pull request merged**